pyaedt/modules/LayerStackup.py

```diff
@@ -143,10 +143,616 @@ def __init__(self, app, layertype="signal", negative=False):
+        """
+        return self.IsMeshOverlay
+
+    @is_mesh_overlay.setter
+    def is_mesh_overlay(self, val):
+        self.IsMeshOverlay = val
+        self.update_stackup_layer()
+
+    @property
+    def locked(self):
+        """Get/Set the active layer lock flag.
+
+        Returns
+        -------
+        bool
+        """
+        return self._locked
+
+    @locked.setter
+    def locked(self, val):
+        self._locked = val
+        self.update_stackup_layer()
+
+    @property
+    def topbottom(self):
+        """Get/Set the active layer top bottom alignment.
+
+        Returns
+        -------
+        str
+        """
+        return self._topbottom
+
+    @topbottom.setter
+    def topbottom(self, val):
+        self._topbottom = val
+        self.update_stackup_layer()
+
+    @property
+    def pattern(self):
+        """Get/Set the active layer pattern.
+
+        Returns
+        -------
+        float
+        """
+        return self._pattern
+
+    @pattern.setter
+    def pattern(self, val):
+        self._pattern = val
+        self.update_stackup_layer()
+
+    @property
+    def drawoverride(self):
+        """Get/Set the active layer draw override value.
+
+        Returns
+        -------
+        float
+        """
+        return self._drawoverride
+
+    @drawoverride.setter
+    def drawoverride(self, val):
+        self._drawoverride = val
+        self.update_stackup_layer()
+
+    @property
+    def thickness(self):
+        """Get/Set the active layer thickness value.
+
+        Returns
+        -------
+        float
+        """
+        return self._thickness
+
+    @thickness.setter
+    def thickness(self, val):
+        self._thickness = val
+        self.update_stackup_layer()
+
+    @property
+    def lowerelevation(self):
+        """Get/Set the active layer lower elevation.
+
+        Returns
+        -------
+        float
+        """
+        return self._lowerelevation
+
+    @lowerelevation.setter
+    def lowerelevation(self, val):
+        self._lowerelevation = val
+        self.update_stackup_layer()
+
+    @property
+    def roughness(self):
+        """Get/Set the active layer roughness (with units).
+
+        Returns
+        -------
+        str
+        """
+        return self._roughness
+
+    @roughness.setter
+    def roughness(self, val):
+        self._roughness = val
+        self.update_stackup_layer()
+
+    @property
+    def botroughness(self):
+        """Get/Set the active layer bottom roughness (with units).
+
+        Returns
+        -------
+        str
+        """
+        return self._botroughness
+
+    @botroughness.setter
+    def botroughness(self, val):
+        self._botroughness = val
+        self.update_stackup_layer()
+
+    @property
+    def toprounghenss(self):
+        """Get/Set the active layer top roughness (with units).
+
+        Returns
+        -------
+        str
+        """
+        return self._toprounghenss
+
+    @toprounghenss.setter
```

Comment:
```suggestion
    @top_roughness.setter
```

Reply:
@MaxJPRey the reason I left those names is to keep the back compatibility. their name was this before and If I change I risk to break it. Any suggestion?

Reply:
Could we have both properties (`top_roughness` and `toprounghenss`)
The old one could return the value of the new one.
For instance for the getter of the old one,
```
@toprounghenss
def toprounghenss (self):
    return self.top_roughness
```

For the setter of the old one:
```
  @top_roughness.setter
  def top_roughness(self, val):
      self._top_roughness= val
      self.update_stackup_layer()
```

And so we don't expose the protected attribute `_toprounghenss` anymore.

Reply:
I've made all exposed properties pythonic

Reply:
@maxcapodi78 Perfect! Thanks